### PR TITLE
docs: Application Info field documentation (#10814)

### LINKED ISSUE #10814: Document the "application info" feature
# Summary

There's an [undocumented field](https://github.com/argoproj/argo-cd/issues/4281#issuecomment-689133740) in the Application spec which allows folks to add additional info to an Application details page.

# Proposal

We could probably have a new page like the [one for external links](https://argo-cd.readthedocs.io/en/stable/user-guide/external-url/).

We should also document the field in the [example application spec](https://github.com/argoproj/argo-cd/blob/master/docs/operator-manual/application.yaml).

### PR DESCRIPTION
**Pull request by Hapshanko**
Closes #10814 

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)). 
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [x] I have added a brief description of why this PR is necessary and/or what this PR solves.

Please see [Contribution FAQs](https://argo-cd.readthedocs.io/en/latest/developer-guide/faq/) if you have questions about your pull-request.

### CODE REVIEW

File: docs/user-guide/extra_info.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    value: >-
+      https://example.com
+```
+![External link](../assets/extra_info-1.png)
+
+The additional information will be visible on the ArgoCD Application details page.
+
+![External link](../assets/extra_info.png)
+
+![External link](../assets/extra_info-2.png)
```
Comment (crenshaw-dev) [inline docs/user-guide/extra_info.md:28]: Would you mind adding alt text?

File: docs/user-guide/extra_info.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# Add extra Application info
```
Comment (crenshaw-dev) [inline docs/user-guide/extra_info.md:1]: Can you.add this page to the mkdocs.yml table of contents?

File: docs/operator-manual/application.yaml [hunk 3]
```diff
@@ -152,7 +152,12 @@ spec:
     # name: in-cluster
     # The namespace will only be set for namespace-scoped resources that have not set a value for .metadata.namespace
     namespace: guestbook
-
+    
+  # Extra information to show in the ArgoCD Application details tab
```
Comment (crenshaw-dev) [inline docs/operator-manual/application.yaml:156]:
```suggestion
  # Extra information to show in the Argo CD Application details tab
```

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]
This is awesome, thank you so much!

File: docs/user-guide/extra_info.md (hunk 2 above)
Comment (Hapshanko) [reply docs/user-guide/extra_info.md:1]: Done!

File: docs/user-guide/extra_info.md (hunk 1 above)
Comment (Hapshanko) [reply docs/user-guide/extra_info.md:28]:
```suggestion
![Manifest example](../assets/extra_info-1.png)

The additional information will be visible on the ArgoCD Application details page.

![App details button](../assets/extra_info.png)

![App details info result](../assets/extra_info-2.png)
```

Is this what you meant by alt text?

**Review by crenshaw-dev — APPROVED** [review]

**Pull request merged**